Fix otel collector metric scraping

twunderlich-grapl opened the pull request:
<!-- Thank you for your contribution to Grapl! Please take some time
to fill out this short template to help us review and understand your
PR. -->

### Which issue does this PR correspond to?
https://github.com/grapl-security/issue-tracker/issues/1011
https://github.com/grapl-security/issue-tracker/issues/1004

<!-- Please provide a link to the GitHub Issue this pull request is
meant to address. -->

### What changes does this PR make to Grapl? Why?
This fixes a few issues with metric scraping from the otel collector
1. The scrape interval was less than the scrape timeout
2. Setting up the metrics pipeline properly
3. Getting the otel collector to use a functioning endpoint for Nomad
<!-- Please describe the functional impact of these changes and the
rationale behind them. This will help us understand your approach to
solving the problem. -->

### How were these changes tested?
make up and then checking the otel-collector logs
<!-- Please describe how these changes were tested. There should be
sufficient detail that reviewers can replicate your testing
procedure. If the procedure is a manual one that's OK, your
description will help us work with you to get it automated. -->

Commit:
Fix broken otel-collector metrics pipeline

1. scrape interval has to be greater than scrape timeout
2. metrics has to be defined as a separate pipeline

Commit:
Fix the metric format and add a comment

Commit:
Set up Nomad endpoints for scraping metrics

For unknown reasons, the dns entry for consul doesn't return a working
IP address that can be used for metric scraping with local grapl. As
such we're using the eth0 IP address which seems to be used consistently
by Nomad and Consul

twunderlich-grapl commented on pulumi/infra/observability_env_vars.py:
```diff
@@ -57,13 +59,14 @@ def otel_config(
     config:
       scrape_configs:
         - job_name: 'nomad-server'
-          scrape_interval: 10s
```
The scrape interval has to be greater than the scrape timeout, otherwise the scraping times out before any scraping is actually conducted.
Essentially these are absolute times, scrape_timeout is not relative to the scrape_interval

twunderlich-grapl commented on pulumi/infra/observability_env_vars.py:
```diff
@@ -57,13 +59,14 @@ def otel_config(
           params:
             format: ['prometheus']
           static_configs:
-            - targets: ['localhost:4646']
```
This doesn't work from inside a container.

christophermaier replied:
I don't want to derail this PR, or the work you've already put into solving this problem (after all, a solution that works is much better than one that doesn't!), but are you familiar with any way of configuring the containers or their networking such that we don't have to jump through so many hoops to get this wired together?

If the answer is "no, this is how we do this", that's fine... it just seems like a lot of fiddly bits to deal with.

jgrillo-grapl reviewed (approved):
LGTM

Commit:
Make env variable required

christophermaier commented on pulumi/infra/config.py:
```diff
@@ -21,6 +21,11 @@
+# This is equivalent to what "${attr.unique.network.ip-address}" resolves to but is used for cases where variable
+# interpolation is not available. Currently this is used to get the private IP for Nomad so the otel collector can
+# scrape metrics
+LOCAL_HOST_IP: Final[str] = os.environ["LOCAL_HOST_ETH0_IP"]
```
Given that this variable is provided in contrast with `HOST_IP_IN_NOMAD`, it may be better to name it something like `HOST_IP_OUTSIDE_NOMAD` (ugh, that name _sucks_ 🤢) to more clearly tie it to the other variable, and to better highlight its intended usage.

Commit:
Use Nomad template syntax to get the private IP

So apparently the templating syntax for Nomad template stanzas is different
than the templating syntax used everywhere else in Nomad. As such I've
rolled back the very hacky get eth0 IP stuff and I'm just using the
native nomad attribute for the host IP WHICH IS WHAT I WANTED ORIGINALLY.
Why does Nomad need at least 3 different syntaxes for variables and
why do certain things only take one of these syntaxes? Ugh Hashicorp
